traits/ctraits.c

```diff
@@ -3466,6 +3466,28 @@ validate_trait_function(
     return raise_trait_error(trait, obj, name, value);
 }
 
+
+/*-----------------------------------------------------------------------------
+|  Verifies a Python value is a callable (or None):
++----------------------------------------------------------------------------*/
+
+static PyObject *
+validate_trait_callable(
+    trait_object *trait, has_traits_object *obj, PyObject *name,
+    PyObject *value)
+{
+
+    PyObject *type_info = trait->py_validate;
+    int kind = PyTuple_GET_SIZE(type_info);
```

Comment:
What's the purpose of the `kind` check? Isn't the size always 1 at this point?

I don't think I understand how this works if `allow_none=False` in the original trait.

Reply:
Looks like `allow_none` is not supported on master, or did I do something stupid? It is true that checking the kind here is a bit pointless here.

Reply:
Agreed; looks like there's no support for `Callable(allow_none=True)` on master. So we should probably remove the `MyCallableCanBeNone` and related tests.

Reply:
Having said that, I could see `allow_none=False` being potentially useful, but I think that's another issue/PR.

Reply:
Removing the related tests sound good to me. They are a bit confusing given the feature hasn't been supported.
:+1: to extending `allow_none` support to `Callable`, in a separate issue/PR.